### ISSUE TITLE
Append a Request Header value from a Secret

### ISSUE THREAD

**Issue by arkodg** [envoyproxy/gateway#4757]

**Comment by arkodg** [comment]
there doesnt seem to be a way to read a header value from sds config source
today https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/base.proto#envoy-v3-api-msg-config-core-v3-headervalueoption

will need something like https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/config_source.proto#config-core-v3-configsource to implement this

**Comment by mathetake** [comment]
alternatively simply EG can read the secret and use the plain header mutation?

**Comment by arkodg** [comment]
The value will be exposed in the config dump which is not desirable

**Comment by mathetake** [comment]
yeah though i imagine people capable of dumping configs == people having access to secrets?

**Comment by arkodg** [comment]
that may not be the case, app devs (xRoute creators) may also hold sensitive info they may not want to share with infra admins (managing EG infra) like basic auth keys, API keys

**Comment by arkodg** [comment]
upstream issue https://github.com/envoyproxy/envoy/issues/37656

**Comment by arkodg** [comment]
in envoy this can we solved using https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/credential_injector_filter

from an EG perspective, this can be added into HTTPRouteFIlter https://gateway.envoyproxy.io/docs/api/extension_types/#httproutefilterspec
cc @missBerg

**Comment by maulik13** [comment]
Hi @arkodg, is there a way to use `credential_injector_filter` in Envoy Gateway using either a patch policy or extensions? (while we wait for having this accessible directly in the GW api?)

Some of our backend services have their own authentication.

**Comment by arkodg** [comment]
Keeping this open to complete docs work
